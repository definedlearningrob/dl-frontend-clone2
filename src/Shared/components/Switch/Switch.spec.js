import { render, fireEvent, screen } from '@testing-library/react';

import SharedSwitch from '@shared/components/Switch/Switch';

const renderSwitch = (props) => {
  const utils = render(<SharedSwitch onChange={() => {}} {...props} />);
  const Switch = utils.getByRole('switch');

  return { ...utils, Switch };
};

describe('Switch', () => {
  it('should render correctly', () => {
    const { container } = renderSwitch({ label: 'Switch label' });
    expect(screen.getByText('Switch label')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('calls "onChange" prop on Switch change', () => {
    const onChange = jest.fn();

    const { Switch } = renderSwitch({ onChange });

    fireEvent.click(Switch);

    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it('disabled state works correctly', () => {
    const { Switch } = renderSwitch({ disabled: true });

    expect(Switch).toBeDisabled();
  });

  it('set switch on when value is true', () => {
    const { Switch } = renderSwitch({ value: true });

    expect(Switch).toBeChecked();
  });

  it('set switch off when value is false', () => {
    const { Switch } = renderSwitch({ value: false });

    expect(Switch).not.toBeChecked();
  });
});
