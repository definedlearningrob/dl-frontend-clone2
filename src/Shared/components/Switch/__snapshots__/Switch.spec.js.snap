// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should render correctly 1`] = `
<div>
  <label
    class="switch"
  >
    <div
      class="content"
    >
      <input
        class="switchInput"
        role="switch"
        type="checkbox"
      />
      <span
        class="slider"
      />
    </div>
    <span
      class="label"
    >
      Switch label
    </span>
  </label>
</div>
`;
